src/test/ui/issues/issue-51632-try-desugar-incompatible-types.stderr

```diff
@@ -1,9 +1,11 @@
-error[E0308]: try expression alternatives have incompatible types
+error[E0308]: `?` operator has incompatible types
   --> $DIR/issue-51632-try-desugar-incompatible-types.rs:8:5
    |
 LL |     missing_discourses()?
    |     ^^^^^^^^^^^^^^^^^^^^^ expected enum `std::result::Result`, found `isize`
    |
+   = note: `?` operator cannot convert from `isize` to `std::result::Result<isize, ()>`
+   = help: `?` operator unwraps `Result` or `Option` and propagates its `Err` or `None`
```

Comment:
It's verbose?

Reply:
In my opinion this last note is not necessary. The clarification with concrete types of the first note should be enough, and people may be using the `try_trait` feature wit custom types where this won't apply.

Reply:
I agree, I feel the second note is probably unnecessary, no-one is likely to try to use `?` without knowing that `Result` and `Option` are supported.